networkx/utils/backends.py

```diff
@@ -122,11 +124,33 @@ def __getitem__(self, name):
 _registered_algorithms = {}
 
 
-def _register_algo(name, wrapped_func):
+def _register_algo(name, wrapped_func, dispatch_args):
     if name in _registered_algorithms:
         raise KeyError(f"Algorithm already exists in dispatch registry: {name}")
     _registered_algorithms[name] = wrapped_func
     wrapped_func.dispatchname = name
+    wrapped_func.dispatchargs = dispatch_args
+
+
+def reregister_all_algos(dispatch_override=None):
+    global _registered_algorithms
+
+    dispatch_fn = _dispatch if dispatch_override is None else dispatch_override
+    algos, _registered_algorithms = _registered_algorithms.copy(), {}
+    _registered_algorithms = {}
```

Comment:
`_registered_algorithms` is set twice here. I suspect you meant to update this to:
```suggestion
    algos = _registered_algorithms.copy()
    _registered_algorithms = {}
```